api/src/main/java/net/kyori/adventure/audience/Audience.java

```diff
@@ -185,157 +191,261 @@ default void forEachAudience(final @NotNull Consumer<? super Audience> action) {
 
   /**
-   * Sends a chat message.
+   * Sends a player chat message from the entity represented by the given {@link Identity} to this {@link Audience} with the {@link ChatType} corresponding to the provided {@link MessageType}.
    *
    * @param source the identity of the source of the message
    * @param message a message
    * @param type the type
    * @see Component
    * @since 4.0.0
+   * @deprecated for removal since 4.12.0, {@link MessageType} is deprecated for removal and the client errors on receiving and can reject identified messages without {@link SignedMessage} data, use {@link #sendMessage(SignedMessage, ChatType.Bound)} instead
    */
+  @ApiStatus.ScheduledForRemoval(inVersion = "5.0.0")
+  @Deprecated
   default void sendMessage(final @NotNull Identity source, final @NotNull Component message, final @NotNull MessageType type) {
+    if (source == Identity.nil()) {
+      this.sendMessage(message);
+    }
+  }
+  /* End: unsigned player messages */
+
+  /* Start: signed player messages */
+  /**
+   * Sends a signed player chat message from the given {@link PlayerIdentified} to this {@link Audience} with the {@link ChatType#CHAT chat} chat type.
+   *
+   * @param signedMessage the signed message data
+   * @param source the source of the message
+   * @since 4.12.0
+   */
+  @ForwardingAudienceOverrideNotRequired
+  default void sendMessage(final @NotNull SignedMessage signedMessage, final @NotNull PlayerIdentified source) {
+    this.sendMessage(signedMessage, source.identity());
+  }
+
+  /**
+   * Sends a signed player chat message from the player identified by the provided {@link PlayerIdentity} to this {@link Audience} with the {@link ChatType#CHAT chat} chat type.
+   *
+   * @param signedMessage the signed message data
+   * @param source the identity of the source of the message
+   * @since 4.12.0
+   */
+  @ForwardingAudienceOverrideNotRequired
+  default void sendMessage(final @NotNull SignedMessage signedMessage, final @NotNull PlayerIdentity source) {
+    this.sendMessage(signedMessage, ChatType.CHAT.bind(source.name()));
+  }
+
+  /**
+   * Sends a signed player chat message to this {@link Audience} with the provided {@link ChatType.Bound} bound chat type.
+   *
+   * @param signedMessage the signed message data
+   * @param boundChatType the bound chat type
+   * @since 4.12.0
+   */
+  default void sendMessage(final @NotNull SignedMessage signedMessage, final ChatType.Bound boundChatType) {
+  }
+
+  /**
+   * Requests deletion of a message with the provided {@link SignedMessage}'s signature.
+   *
+   * @param signedMessage the message to delete
+   * @see SignedMessage#canDelete()
+   * @since 4.12.0
+   * @sinceMinecraft 1.19
+   */
+  @ForwardingAudienceOverrideNotRequired
+  default void deleteMessage(final @NotNull SignedMessage signedMessage) {
+    if (signedMessage.canDelete()) {
+      this.deleteMessage(Objects.requireNonNull(signedMessage.signature(), "signedMessage must have a signature"));
```

Comment:
`SignedMessage#canDelete` already does a null check on the signature, so this null check is redundant. Is this intentional due to possible future changes to the canDelete method?

Reply:
I *think* I just put that there to tell the IDE to stop complaining, but it was a while ago so idk for sure.